include/libpy/err.h

```diff
@@ -200,6 +202,21 @@ msgbuilder raise(exctype type);
 void raise(py::err::object err);
 
 /**
+   Raise a Python exception from a C++ exception and then throw the C++
+   exception.
+
+   @param exc The C++ exception to raise a Python exception from and throw.
+   @param type The type of Python exception to raise.
+*/
+template<typename T,
+         typename = std::enable_if_t<std::is_base_of<std::exception, T>::value>>
+[[noreturn]] void raise(T &&err, exctype type = RuntimeError) {
```

Comment:
👍 on this helper and on the many helpful language/stdlib features I looked up while reading this (`std::enable_if_t`, `std::is_base_of`, `[[noreturn]]`, `std::exception::what`).